src/ui/base-page.js

```diff
@@ -0,0 +1,259 @@
+                options: {
+                    model:{
+                        ttsText: null,
+                        utteranceOpts: {
+                            lang: "{page}.model.uiLanguage"
+                        }
+                    },
+                    listeners: {
+                        "{page}.events.onStoryListenToRequested": {
+                            func: "{that}.queueSpeech",
+                            args: ["{that}.model.ttsText", true],
+                            namespace: "queueSpeech"
+                        }
+                    }
+                }
+            },
+            // the storytelling tool "main" menu
+            menu: {
+                type: "sjrk.storyTelling.ui.menu",
+                container: ".sjrkc-st-menu"
+            },
+            uio: {
+                type: "fluid.uiOptions.prefsEditor.multilingualDemo",
+                container: ".flc-prefsEditor-separatedPanel",
+                options: {
+                    components: {
+                        prefsEditorLoader: {
+                            options: {
+                                components: {
+                                    prefsEditor: {
+                                        options: {
+                                            listeners: {
+                                                "onCreate.escalate": "{page}.events.onUioReady.fire",
+                                                "{messageLoader}.events.onResourcesLoaded": [{
+                                                    func: "{that}.events.onPrefsEditorRefresh",
+                                                    namespace: "rerenderUIO"
+                                                },
+                                                {
+                                                    func: "{page}.events.onUioReady.fire",
+                                                    namespace: "escalate"
+                                                },
+                                                {
+                                                    funcName: "sjrk.storyTelling.base.page.updateUioPanelLanguages",
+                                                    args: ["{prefsEditorLoader}", "{page}"],
+                                                    priority: "before:rerenderUIO",
+                                                    namespace: "updateMessageBases"
+                                                }]
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    sjrk.storyTelling.base.page.renderAllUiTemplates = function (component) {
+        fluid.each(component, function (subcomponent) {
```

Comment:
This is messy, instead create a renderAllUITemplates event in the component and distribute down an interaction grade to the ui components which listens to is
e.g. 
fluid.defaults("sjrk.storyTelling.uiInBasePage", {
    listeners: {
        "{sjrk.storyTelling.base.page}.events.renderAllUiTemplates": "templateManager.events.onResourceLoadRequested.fire"
    }
}
and then in the base page
distributeOptions: {
    target: "{that sjrk.storyTelling.ui}.options.gradeNames",
    record: "sjrk.storyTelling.uiInBasePage"
    }